pkg/network/usm/utils/connection_rollup_test.go

```diff
@@ -0,0 +1,136 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package utils
+
+import (
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/network/types"
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRollupKey(t *testing.T) {
+	t.Run("same key", func(t *testing.T) {
+		aggregator := NewConnectionAggregator()
+		srcIP := util.AddressFromString("1.1.1.1")
+		dstIP := util.AddressFromString("2.2.2.2")
+
+		c1 := types.NewConnectionKey(srcIP, dstIP, 6000, 80)
+		t1 := aggregator.RollupKey(c1)
+		t2 := aggregator.RollupKey(c1)
+
+		assert.Equal(t, c1, t1)
+		assert.Equal(t, c1, t2)
+	})
+
+	t.Run("same key, flipped order", func(t *testing.T) {
+		aggregator := NewConnectionAggregator()
+		srcIP := util.AddressFromString("1.1.1.1")
+		dstIP := util.AddressFromString("2.2.2.2")
+
+		c1 := types.NewConnectionKey(srcIP, dstIP, 6000, 80)
+		t1 := aggregator.RollupKey(c1)
+
+		c2 := types.NewConnectionKey(dstIP, srcIP, 80, 6000)
+		t2 := aggregator.RollupKey(c2)
+
+		assert.Equal(t, c1, t1)
+		assert.Equal(t, c1, t2)
+	})
+
+	t.Run("same IPs, but no matching ports", func(t *testing.T) {
+		aggregator := NewConnectionAggregator()
+		srcIP := util.AddressFromString("1.1.1.1")
+		dstIP := util.AddressFromString("2.2.2.2")
+
+		c1 := types.NewConnectionKey(srcIP, dstIP, 6000, 80)
+		t1 := aggregator.RollupKey(c1)
+
+		c2 := types.NewConnectionKey(dstIP, srcIP, 7000, 53)
+		t2 := aggregator.RollupKey(c2)
+
+		// In this case both keys are preserved, which wouldn't trigger a rollup
+		assert.Equal(t, c1, t1)
+		assert.Equal(t, c2, t2)
```

Comment:
maybe lets add `assert.NoEqual(t, c1, c2)` to validate we we're not missing anything?

Reply:
👍 Done